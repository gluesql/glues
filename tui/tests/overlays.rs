#[macro_use]
mod tester;
use tester::Tester;

use color_eyre::Result;

#[tokio::test]
async fn keymap_toggles() -> Result<()> {
    let mut t = Tester::new().await?;
    t.open_instant().await?;

    // show keymap
    t.press('?').await;
    t.draw()?;
    snap!(t, "keymap_shown");

    // hide keymap
    t.press('?').await;
    t.draw()?;
    snap!(t, "keymap_hidden");

    Ok(())
}

