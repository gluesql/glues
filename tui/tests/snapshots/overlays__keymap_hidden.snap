---
source: tui/tests/overlays.rs
expression: text
snapshot_kind: text
---
 Directory 'Notes' selected                                                                            [?] Show keymap 
[Browser]                                   ▐[Editor]                                                                   
 󰝰 Notes                                    ▐ 1 Welcome to Glues :D                                                     
   󱇗 Sample Note                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐                                                                           
                                            ▐
